interfaces/builtin/serial_port.go

```diff
@@ -67,20 +82,42 @@ func (iface *SerialPortInterface) SanitizeSlot(slot *interfaces.Slot) error {
 }
 
 // SanitizePlug checks and possibly modifies a plug.
-func (iface *SerialPortInterface) SanitizePlug(slot *interfaces.Plug) error {
-	if iface.Name() != slot.Interface {
+func (iface *SerialPortInterface) SanitizePlug(plug *interfaces.Plug) error {
+	if iface.Name() != plug.Interface {
 		panic(fmt.Sprintf("plug is not of interface %q", iface))
 	}
-	// NOTE: currently we don't check anything on the plug side.
+
```

Comment:
We should verify here too for the case that we have a path specified that we're only allowed to connect when the plug as the path attribute with the same value specified.

To improve the user experience for this we may should add attribtes to the output of `snap interfaces`